python_modules/dagster/dagster/_core/definitions/freshness_checks/last_update.py

```diff
@@ -100,10 +134,166 @@ def build_last_update_freshness_checks(
+                    if deadline_cron
+                    else check.not_none(update_timestamp) + lower_bound_delta.total_seconds()
+                )
+                metadata[FRESH_UNTIL_METADATA_KEY] = TimestampMetadataValue(fresh_until)
+            if update_timestamp:
+                metadata[LAST_UPDATED_TIMESTAMP_METADATA_KEY] = TimestampMetadataValue(
+                    update_timestamp
+                )
+
+            yield AssetCheckResult(
+                passed=passed,
+                description=_construct_description(
+                    passed,
+                    last_update_time_lower_bound=last_update_time_lower_bound.timestamp(),
+                    current_timestamp=current_timestamp,
+                    update_timestamp=update_timestamp,
+                    out_of_date_observation=(
+                        is_observation_out_of_date(latest_record, last_update_time_lower_bound)
+                        if latest_record
+                        else False
+                    ),
+                    latest_event_type=(latest_record.event_type if latest_record else None),
+                ),
+                severity=severity,
+                asset_key=asset_key,
+                metadata=metadata,
+            )
+
+    return the_check
+
+
+def _construct_description(
+    passed: bool,
+    last_update_time_lower_bound: Optional[float],
+    current_timestamp: float,
+    update_timestamp: Optional[float],
+    out_of_date_observation: bool,
+    latest_event_type: Optional[DagsterEventType],
+) -> str:
+    check.invariant(
+        (passed and update_timestamp is not None) or not passed,
+        "Should not be possible for check to pass without an update to the asset.",
+    )
+
+    record_type = (
+        "materialization"
+        if latest_event_type == DagsterEventType.ASSET_MATERIALIZATION
+        else "observation"
+    )
+    allowed_delta = (
+        seconds_in_words(current_timestamp - last_update_time_lower_bound)
+        if last_update_time_lower_bound
+        else None
+    )
+    actual_delta = (
+        seconds_in_words(current_timestamp - update_timestamp) if update_timestamp else None
+    )
+
+    if passed:
+        return f"Asset is currently fresh. The latest {record_type} was {actual_delta} ago, within the allowed time range of {allowed_delta}."
+    elif update_timestamp is None:
+        return "Asset has never been observed/materialized."
+    elif out_of_date_observation:
+        return (
+            "Asset is in an unknown state. No run has observed the asset recently enough "
+            f"(within the last {allowed_delta}) for us to know if it is fresh or overdue. Because of "
+            "this ambiguity, we marked the check as failed."
+        )
+    else:
+        return (
+            f"Asset is overdue for an update. The latest {record_type} was {actual_delta} ago, "
+            f"which is past the allowed time range of {allowed_delta}."
```

Comment:
Nitpick: we typically use "range" to refer to a tuple of (start, end). "delta" or "duration" would be more accurate.